File: TP/TP_python/enseignant.py
from personne import *

class Enseignant(Personne):
    def __init__(self, nom, age, salaire, matiere):
        super().__init__(nom, age)
        self.salaire = salaire
        self.matiere = matiere

    def afficher_details(self):
        super().afficher_details()
        print(f"Salaire: {self.salaire}, Matière: {self.matiere}")

    

File: src/main/java/Enseignant.java
public class Enseignant extends Personne {
    private String matiere;
    private double salaire;

    public Enseignant(String nom, int age, String matiere, double salaire) {
        super(nom, age);
        this.matiere = matiere;
        this.salaire = salaire;
    }

    @Override
    public String afficherDetails() {
        return super.afficherDetails() + " | Enseignant en " + matiere + " (Salaire: " + salaire + "euros)";
    }
}